### PR TITLE
Don't mutate the global VTTCue

### PR DESCRIPTION
**Pull request by forbesjo**

### CODE REVIEW

File: lib/media/text_engine.js [hunk 1]
```diff
@@ -110,7 +110,47 @@ shaka.media.TextEngine.makeCue = function(startTime, endTime, payload) {
...
+    return new window.TextTrackCue(startTime, endTime, payload);
+
+  // Draft spec TextTrackCue with 6 constructor arguments (5th & 6th are
+  // optional).
+  // See {@link https://goo.gl/AYFqUh W3C Working Draft 29 March 2012}.
+  // Quoting the access to the TextTrackCue object to avoid the compiler
+  // complaining.
+  } else if (constructorLength == 6) {
+    var id = startTime + '-' + endTime + '-' + payload;
+    return new window['TextTrackCue'](id, startTime, endTime, payload);
```
Comment (gesinger) [inline lib/media/text_engine.js:145]: Any specific reason for using brackets instead of accessing the property directly?

**Review by gesinger — COMMENTED** [review]
May want to remove the line in `docs/tutorials/plugins.md` where polyfills are mentioned, and maybe add a unit test that ensures that the polyfills don't overwrite an already polyfilled VTTCue (in case some merge in the future tries to bring it back).

File: lib/media/text_engine.js (hunk 1 above)
Comment (forbesjo) [reply lib/media/text_engine.js:145]: It's from the original polyfill https://github.com/google/shaka-player/blob/master/lib/polyfill/vttcue.js#L91, Looking through the history I don't see a reason for why this is using brackets.

**Comment by forbesjo** [comment]
Removed doc note and updated tests.

**Comment by forbesjo** [comment]
Confirmed that this method isn't too slow and doesn't impact performance noticeably.

File: test/media/vtt_text_parser_unit.js [hunk 2]
```diff
@@ -46,6 +47,9 @@ describe('VttTextParser', function() {
     verifyHelper([],
         'WEBVTT',
         { periodStart: 0, segmentStart: 0, segmentEnd: 0 });
+
+    // window.VTTCue is not overwritten
+    expect(window.VTTCue.toString()).toBe('function () {}');
```
Comment (gesinger) [inline test/media/vtt_text_parser_unit.js]: Although this should work in most cases, it's possible it wouldn't catch an instance where VTTCue is overwritten by another empty function. Instead maybe just store a reference to a function, set `window.VTTCue = funcReference` in `beforeEach`, then test that `window.VTTCue` is the same as the reference var in the test.

Comment (forbesjo) [reply test/media/vtt_text_parser_unit.js]: Sounds good

**Review by gesinger — APPROVED** [review]

**Pull request merged**